internal/services/disbursement_management_service.go

```diff
@@ -39,16 +39,33 @@ type DisbursementWithUserMetadata struct {
 }
 
 var (
-	ErrDisbursementNotFound                  = errors.New("disbursement not found")
-	ErrDisbursementNotReadyToStart           = errors.New("disbursement is not ready to be started")
-	ErrDisbursementNotReadyToPause           = errors.New("disbursement is not ready to be paused")
-	ErrDisbursementWalletDisabled            = errors.New("disbursement wallet is disabled")
-	ErrDisbursementWalletInsufficientBalance = errors.New("disbursement wallet has insufficient balance to fulfill disbursement and current pending payments")
+	ErrDisbursementNotFound        = errors.New("disbursement not found")
+	ErrDisbursementNotReadyToStart = errors.New("disbursement is not ready to be started")
+	ErrDisbursementNotReadyToPause = errors.New("disbursement is not ready to be paused")
+	ErrDisbursementWalletDisabled  = errors.New("disbursement wallet is disabled")
 
 	ErrDisbursementStatusCantBeChanged = errors.New("disbursement status can't be changed to the requested status")
 	ErrDisbursementStartedByCreator    = errors.New("disbursement can't be started by its creator")
 )
 
+type InsufficientBalanceError struct {
+	DisbursementID     string
+	AvailableBalance   float64
+	DisbursementAmount float64
+	TotalPendingAmount float64
+}
+
+func (e InsufficientBalanceError) Error() string {
+	return fmt.Sprintf(
+		"disbursement %s failed with insufficient distribution account balance %f to fulfill new amount (%f) along with the pending amount (%f). Total pending amount: %f",
+		e.DisbursementID,
+		e.AvailableBalance,
+		e.DisbursementAmount,
+		e.TotalPendingAmount,
+		(e.DisbursementAmount+e.TotalPendingAmount)-e.AvailableBalance,
```

Comment:
Can we change `Total pending amount: %f` to something like `distribution account top up needed` or smth? I wound't be able to understand what this last number is just from the phrasing.

Reply:
Updated to the following format:
![Screenshot 2024-02-09 at 12 42 26 PM](https://github.com/stellar/stellar-disbursement-platform-backend/assets/1952597/e9817e47-9e34-4555-ba54-8743cb979911)